workos/types/user_management/authenticate_with_common.py

```diff
@@ -0,0 +1,63 @@
+from typing import Literal, TypedDict, Union
+
+
+class AuthenticateWithBaseParameters(TypedDict):
+    ip_address: Union[str, None]
+    user_agent: Union[str, None]
+
+
+class AuthenticateWithPasswordParameters(AuthenticateWithBaseParameters):
+    email: str
+    password: str
+    grant_type: Literal["password"]
+
+
+class AuthenticateWithCodeParameters(AuthenticateWithBaseParameters):
+    code: str
+    code_verifier: Union[str, None]
+    grant_type: Literal["authorization_code"]
+
+
+class AuthenticateWithMagicAuthParameters(AuthenticateWithBaseParameters):
+    code: str
+    email: str
+    link_authorization_code: Union[str, None]
+    grant_type: Literal["urn:workos:oauth:grant-type:magic-auth:code"]
+
+
+class AuthenticateWithEmailVerificationParameters(AuthenticateWithBaseParameters):
+    code: str
+    pending_authentication_token: str
+    grant_type: Literal["urn:workos:oauth:grant-type:email-verification:code"]
+
+
+class AuthenticateWithTotpParameters(AuthenticateWithBaseParameters):
+    code: str
+    authentication_challenge_id: str
+    pending_authentication_token: str
+    grant_type: Literal["urn:workos:oauth:grant-type:mfa-totp"]
+
+
+class AuthenticateWithOrganizationSelectionParameters(AuthenticateWithBaseParameters):
+    organization_id: str
+    pending_authentication_token: str
+    grant_type: Literal["urn:workos:oauth:grant-type:organization-selection"]
+
+
+class AuthenticateWithRefreshTokenParameters(AuthenticateWithBaseParameters):
+    client_id: str
+    client_secret: str
+    refresh_token: str
+    organization_id: Union[str, None]
+    grant_type: Literal["refresh_token"]
+
+
+AuthenticateWithParameters = Union[
+    AuthenticateWithPasswordParameters,
+    AuthenticateWithCodeParameters,
+    AuthenticateWithMagicAuthParameters,
+    AuthenticateWithEmailVerificationParameters,
+    AuthenticateWithTotpParameters,
+    AuthenticateWithOrganizationSelectionParameters,
+    # AuthenticateWithRefreshTokenParameters is purposely omitted due to its divergent response typing
```

Comment:
There are input types, what are you referring to with "divergent response typing"?

Reply:
This should actually say:
>AuthenticateWithRefreshTokenParameters is purposely omitted from this union because it doesn't use the authenticate_with() method due to its divergent response typing